rmgpy/solver/simple.pyx

```diff
@@ -253,20 +258,34 @@ cdef class SimpleReactor(ReactionSystem):
                 
     def get_threshold_rate_constants(self, modelSettings):
         """
-        Get the threshold rate constants for reaction filtering.
+        Get the threshold rate constants for reaction filtering by setting the maximum uni-/bimolecular rate as a
+        custom rate constant thresholds for each reaction family.
         """
-        # Set the maximum unimolecular rate to be kB*T/h
-        unimolecular_threshold_rate_constant = 2.08366122e10 * self.T.value_si
-        # Set the maximum bi/trimolecular rate by using the user-defined rate constant threshold
-        bimolecular_threshold_rate_constant = modelSettings.filterThreshold
-        # Maximum trimolecular rate constants are approximately three
-        # orders of magnitude smaller (accounting for the unit
-        # conversion from m^3/mol/s to m^6/mol^2/s) based on
-        # extending the Smoluchowski equation to three molecules
-        trimolecular_threshold_rate_constant = modelSettings.filterThreshold / 1e3
-        return (unimolecular_threshold_rate_constant,
-                bimolecular_threshold_rate_constant,
-                trimolecular_threshold_rate_constant)
+
+        # Evaluate kinetics at user defined temperature
+        # unimolecular reactions
+        kvals_uni_tmp = []
+        for k, unimol_kinetics in enumerate(modelSettings.unimolecularFilterFit):
+            if unimol_kinetics:
+                kvals_uni_tmp.append(min(unimol_kinetics.getRateCoefficient(self.T.value_si),
+                                         constants.kB * self.T.value_si))
+            else:
+                # Set the maximum unimolecular rate to be kB*T/h
+                kvals_uni_tmp.append(constants.kB * self.T.value_si)
+
+        # bimolecular reactions
+        kvals_bi_tmp = []
+        for k, bimol_kinetics in enumerate(modelSettings.bimolecularFilterFit):
+            if bimol_kinetics:
+                kvals_bi_tmp.append(min(1.0e8, bimol_kinetics.getRateCoefficient(self.T.value_si)))
+            else:
+                # Value estimated, based on the same order of magnitude of hydrogen colliding at 1000 K.
+                kvals_bi_tmp.append(1e8)
+
+        kvals_uni = numpy.asarray(kvals_uni_tmp, dtype=numpy.float64)
+        kvals_bi = numpy.asarray(kvals_bi_tmp, dtype=numpy.float64)
+
+        return (kvals_uni, kvals_bi)
```

Comment:
We should return a value for trimolecular reactions here instead of using kvals_bi directly in base.pyx.

Additionally, the treatment of trimolecular families is a bit confusing. We should discuss.